flexmeasures/ui/crud/users.py

```diff
@@ -52,48 +37,68 @@ def render_user(user: Optional[User], msg: str = None):
+    user_data: dict, user_id: Optional[int] = None, make_obj=False
+) -> Union[User, dict]:
+    """
+    Turn data from the internal API into something we can use to further populate the UI.
+    Either as an user object or a dict for form filling.
+    """
+    with db.session.no_autoflush:
+        role_ids = tuple(user_data["flexmeasures_roles"])
+        user_data["flexmeasures_roles"] = Role.query.filter(Role.id.in_(role_ids)).all()
+        user_data.pop("status", None)  # might have come from requests.response
+        if user_id:
+            user_data["id"] = user_id
+        if make_obj:
+            user = User(**user_data)
+            db.session.expunge(user)
+            return user
+    return user_data
+
+
+class UserCrudUI(FlaskView):
     route_base = "/users"
     trailing_slash = False
 
     @roles_required("admin")
     def index(self):
         """/users"""
-        only_active = request.args.get("include_inactive", "0") == "0"
-        users = get_users(only_active=only_active)
+        include_inactive = request.args.get("include_inactive", "0") != "0"
+        get_users_response = InternalApi().get(
+            url_for(
+                "flexmeasures_api_v2_0.get_users", include_inactive=include_inactive
+            )
+        )
+        users = [
+            process_internal_api_response(user, make_obj=True)
+            for user in get_users_response.json()
+        ]
         return render_flexmeasures_template(
-            "crud/users.html", users=users, include_inactive=not only_active
+            "crud/users.html", users=users, include_inactive=include_inactive
         )
 
     @roles_required("admin")
     def get(self, id: str):
         """GET from /users/<id>"""
-        user: User = get_user(id)
-        return render_user(user)
-
-    @roles_required("admin")
-    def delete_with_data(self, id: str):
-        """Delete via /users/delete_with_data/<id>"""
-        user: User = get_user(id)
-        username = user.username
-        delete_user(user)
-        return render_user(
-            None,
-            msg="User %s and assorted assets/readings have been deleted." % username,
+        get_user_response = InternalApi().get(
+            url_for("flexmeasures_api_v2_0.get_user", id=id)
         )
+        user = process_internal_api_response(get_user_response.json(), make_obj=True)
+        return render_user(user)
 
     @roles_required("admin")
     def toggle_active(self, id: str):
         """Toggle activation status via /users/toggle_active/<id>"""
         user: User = get_user(id)
-        toggle_activation_status_of(user)
+        InternalApi().patch(
+            url_for("flexmeasures_api_v2_0.patch_user", id=id),
+            args={"active": not user.active},
+        )
         return render_user(
             user,
             msg="User %s's new activation status is now %s."
-            % (user.username, user.active),
+            % (user.username, not user.active),
```

Comment:
We would be more sure of this if we explicitly get the user again after patching.